src/nemos/observation_models.py

```diff
@@ -1165,3 +1165,296 @@ def estimate_scale(
             The DOF of the residuals.
         """
         return jnp.ones_like(jnp.atleast_1d(y[0]))
+
+
+class NegativeBinomialObservations(Observations):
+    r"""
+    A Negative Binomial model for overdispersed count data using mean-dispersion parameterization.
+
+    This model represents a Negative Binomial distribution [3]_ commonly used to model overdispersed
+    count data [4]_ [5]_ (i.e., data where the variance exceeds the mean), which cannot be captured by a
+    standard Poisson model. The distribution is parameterized by the predicted mean rate
+    (:math:`\mu`) and a fixed dispersion parameter (:math:`\phi`) or "scale" of the model.
```

Comment:
```suggestion
    (:math:`\mu`) and a fixed dispersion parameter (:math:`\phi`) or `scale` of the model.
```

since this is the parameter they're setting